Tasks/AzureRmWebAppDeployment/operations/AzureResourceFilterUtility.ts

```diff
@@ -0,0 +1,22 @@
+import tl = require('vsts-task-lib/task');
+import { AzureEndpoint } from 'azure-arm-rest/azureModels';
+import { Resources } from 'azure-arm-rest/azure-arm-resource';
+
+export class AzureResourceFilterUtility {
+    public static async getResourceGroupName(endpoint: AzureEndpoint, resourceType: string, resourceName: string): Promise<string> {
+        var azureResources: Resources = new Resources(endpoint);
+        var filteredResources: Array<any> = await azureResources.getResources(resourceType, resourceName);
+        let resourceGroupName: string;
+        if(!filteredResources || filteredResources.length == 0) {
+            throw new Error(tl.loc('ResourceDoesntExist', resourceName));
+        }
+        else if(filteredResources.length == 1) {
+            resourceGroupName = filteredResources[0].id.split("/")[4];
+        }
+        else {
+            throw new Error(tl.loc('MultipleResourceGroupFoundForAppService', resourceName));
```

Comment:
Should we take resource type as input to log string.

Reply:
`Multiple resource group found for App Service '%s'`  seems to be explicit for users.
We should be good here.

Reply:
done